README.md

```diff
@@ -60,6 +60,190 @@ $$
 function_info = client.create_function(sql_function_body=sql_body)
 ```
 
+#### Create a Unity Catalog Python Function
+
+The `create_python_function` API allows you to directly convert a Python function into a Unity Catalog (UC) User-Defined Function (UDF). It automates the creation of UC functions, while ensuring the Python function meets certain criteria and adheres to best practices.
+
+##### Requirements and Guidelines
+
+**Type Annotations**:
+
+- **Mandatory**: The Python function must use argument and return type annotations. These annotations are used to generate the SQL signature of the UC function.
+- **Supported Types**: The following Python types are supported and mapped to their corresponding Unity Catalog SQL types:
+
+    | Python Type         | Unity Catalog Type |
+    |---------------------|--------------------|
+    | `int`               | `INTEGER`          |
+    | `float`             | `DOUBLE`           |
+    | `str`               | `STRING`           |
+    | `bool`              | `BOOLEAN`          |
+    | `Decimal`           | `DECIMAL`          |
+    | `datetime.date`     | `DATE`             |
+    | `datetime.datetime` | `TIMESTAMP`        |
+    | `list`              | `ARRAY`            |
+    | `tuple`             | `ARRAY`            |
+    | `dict`              | `MAP`              |
+
+- **Container Types**: If you use container types like `list`, `tuple`, or `dict`, you **must specify the inner types** explicitly. Unity Catalog does not support ambiguous types like `List` or `Dict`. For example:
+
+```python
+def process_data(ids: List[int], details: Dict[str, float]) -> List[str]:
```

Comment:
Is a function with optional values supported?

```
def f(s: str = None)
```

Reply:
it is now :) Thanks for the call-out!!